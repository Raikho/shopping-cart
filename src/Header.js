import { NavLink } from 'react-router-dom';

const Header = props => {
    return (
        <div className="header">
            <div className="store-title">ClothesShop</div>
            <div className="nav-container">
                <NavLink to='/' >Home</NavLink>
                <NavLink to='shop/all' >Shop</NavLink>
                <NavLink to='contact' >Contact</NavLink>
            </div>
            <div className="cart-button">Cart</div>
        </div>
    )
}

export default Header;