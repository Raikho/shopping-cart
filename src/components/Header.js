import { NavLink } from 'react-router-dom';

const Header = props => {
    return (
        <div className="header">
            <div className="store-title">Fish Supply</div>
            <div className="nav-container">
                <NavLink to='/' >Home</NavLink>
                <NavLink to='shop/all' >Shop</NavLink>
                <NavLink to='contact' >Contact</NavLink>
            </div>
            <NavLink to='cart' className="cart-button">
                Cart: (${props.total}.00)
            </NavLink>
            {/* <div className="cart-button">Cart: (${props.total}.00)</div> */}
        </div>
    )
}

export default Header;