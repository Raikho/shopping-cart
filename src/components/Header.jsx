export default function Header({ title, total }) {
    return (
        <div className="header">
            <div className="store-title">{title}</div>
            <div className="nav-container">
                <a href="">Home</a>
                <a href="">Shop</a>
                <a href="">Contact</a>
            </div>
            <a href="" data-testid='cart-link'>
                Cart: (${total}.00)
            </a>
        </div>
    );
}