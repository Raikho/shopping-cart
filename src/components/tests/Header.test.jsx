import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import Header from '../Header';

describe('Header component shows links', () => {
    it('renders', () => {
        const { container } = render(<Header total={0}/>);
        expect(container).toMatchSnapshot();
    });
    it('renders store title', () => {
        render(<Header title="Fish Supply" />);
        expect(screen.getByText('Fish Supply')).toBeInTheDocument();
    });
    it('renders navlinks', () => {
        render(<Header total={0}/>);
        expect(screen.getByRole('link', {name: 'Home'})).toBeInTheDocument();
        expect(screen.getByRole('link', {name: 'Shop'})).toBeInTheDocument();
        expect(screen.getByRole('link', {name: 'Contact'})).toBeInTheDocument();
    });
    it('renders cart with correct number', () => {
        render(<Header total={11} />);
        let cart = screen.getByTestId('cart-link');
        expect(cart.textContent).toMatch('Cart: ($11.00)');
    });
});
