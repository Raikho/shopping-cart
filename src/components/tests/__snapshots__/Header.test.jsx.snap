// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component shows links > renders 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="store-title"
    />
    <div
      class="nav-container"
    >
      <a
        href=""
      >
        Home
      </a>
      <a
        href=""
      >
        Shop
      </a>
      <a
        href=""
      >
        Contact
      </a>
    </div>
    <a
      data-testid="cart-link"
      href=""
    >
      Cart: ($
      0
      .00)
    </a>
  </div>
</div>
`;
